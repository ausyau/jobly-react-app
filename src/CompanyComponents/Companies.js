/* Companies component renders a list of companies */

import React, { Component } from 'react';
import JoblyApi from '../JoblyApi';
import { Redirect } from 'react-router-dom';
import UserContext from '../HelperComponents/userContext';
import Companycard from '../CompanyComponents/Companycard';
import SearchForm from '../HelperComponents/SearchForm';


class Companies extends Component {
  static contextType = UserContext;
  constructor(props) {
    super(props);

    this.state = {
      companies: []
    };

    this.getCompanies = this.getCompanies.bind(this);
  }

  async componentDidMount() {
    if (this.context.username) {
      let result = await JoblyApi.getCompanies();
      this.setState({ companies: result });
    }
  }

  componentWillUnmount() {
    this.setState({ companies: [] });
  }

  async getCompanies(search) {
    let result = await JoblyApi.getCompanies(search);
    this.setState({ companies: result });
  }

  render() {

    let loggedIn = <div>
      <SearchForm id="company" search={this.getCompanies} />
      {this.state.companies.map(company => (<Companycard key={company.handle} handle={company.handle} {...company} />))}
    </div>;

    const currentUser = this.context;

    return (
      <div className="Home">
        {currentUser.username
          ? <div>{loggedIn} </div>
          : <Redirect to='/' />
        }
      </div>
    );


  }
}


export default Companies;
