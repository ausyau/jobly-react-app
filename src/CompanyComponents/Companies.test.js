import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Companies from "./Companies";

it("renders without crashing", function () {
  shallow(<Companies />);
});

it("matches snapshot", function () {
  let wrapper = shallow(<Companies />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
