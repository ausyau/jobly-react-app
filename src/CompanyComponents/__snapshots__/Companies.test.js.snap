// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <SearchForm
    id="company"
    search={[Function]}
  />
</div>
`;
